src/messages/MClientRequest.h

```diff
@@ -160,7 +162,17 @@ class MClientRequest : public Message {
 
   void decode_payload() {
     bufferlist::iterator p = payload.begin();
-    ::decode(head, p);
+
+    if (header.version >= 4) {
+      struct ceph_mds_request_head_new	new_mds_head;
+
+      ::decode(new_mds_head, p);
+      memcpy(&head, &(new_mds_head.oldest_client_tid), sizeof(head));
```

Comment:
Eww. We normally do this the other way around: switch to the new structure natively, and instantiate the old one for decoding/encoding if needed. Granted this works but it might not if we change head again. 
(Also we can use the ceph_mds_request_head name for our new structure and just append _old to the classic one that way.)